packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart

```diff
@@ -282,7 +286,7 @@ class IconTreeShaker {
         'got $constFinderMap.');
     }
     final _ConstFinderResult constFinderResult = _ConstFinderResult(constFinderMap);
-    if (constFinderResult.hasNonConstantLocations) {
+    if (constFinderResult.hasNonConstantLocations && _targetPlatform != TargetPlatform.web_javascript) {
```

Comment:
What if there are non-constant locations outside of where the `staticIconProvider` is used? Is totally silencing this for web safe?

Reply:
You're right, this is not safe if the user has created non-const instances of `IconData`. I put this hack because for some reason const constructor declarations were showing up as non-const instances in the web dills, let me investigate further and see if there's a robust way to not count these as non-const instances.

Reply:
removed this hack after https://github.com/flutter/engine/pull/38131